### ISSUE TITLE
Vueform FileElement in ListElement causes "No File Selected" error

### ISSUE THREAD

**Issue by pbrink21** [vueform/vueform#334]
### Environment

* Npm
* Framework: Vite 5.0.10
* Theme: Vueform

### Reproduction

https://stackblitz.com/edit/github-owuwp1?file=src%2FApp.vue

Just create a vueform with a list element with a file element inside of it

### Describe the bug

When you create a form like this, it seems to work until you try to upload an image, then it throws the error "No file selected" when you select a file.

**Comment by adamberecz** [comment]
The `FileElement` should receive the `index` as `name`:
https://stackblitz.com/edit/github-owuwp1-k9rmpr?file=src%2FApp.vue

**Comment by adamberecz** [comment]
Closing this as it doesn't seem to be an issue. If there's something I overlooked feel free to reopen.

**Comment by pbrink21** [comment]
Good to know there's a way to make it work!

I'm actually going to re-open it though, because IMO that's not actually a solution for a few reasons:

1. It doesn't match the functionality of the rest of VueForm elements here (any other one would be able to not be at the index. When I was using this I wanted each file to be accessible from `item.imageName`, not `item[index]`, and it seems confusing and very complicated to have to access things via `[index]` when everything else can be accessed through a `.property`

2. What I was actually trying to do with `ListElement` when I ran into this was slightly more complicated than the example I provided, and your solution does not actually allow it. What I had wanted (and why I wasn't just using `MultiFileElement`) was a list of pairs of images in the form of `{before: image, after: image}`. I tried doing the thing you mentioned with index for this, but here's the problem: if you have anything other than the index in the name, you run into the issue described, and having them both with the same name does not work. Here's an updated example: 
[https://stackblitz.com/edit/github-owuwp1-w62wd4?file=src%2FApp.vue](https://stackblitz.com/edit/github-owuwp1-w62wd4?file=src%2FApp.vue
)
